File: src/Backend/Backend/Models/Dto/EventPinDto.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace Backend.Models.Dto
{
    public class EventPinDto
    {
        public int Id { get; set; }
        public int EventId { get; set; }

        public double Latitude { get; set; }
        public double Longitude { get; set; }

        public string Label { get; set; }
        public string? Description { get; set; }

        public DateTime PinnedAt { get; set; }

        public PinTypes PinCategory { get; set; }
    }
}

File: src/WebApp/src/app/Models/EventPinDto.ts
export class EventPinDto{

    constructor(
        private eventId : number,
        private latitude: number,
        private longitude: number,
        private label : string,
        private pinnedAt : Date,
        private pinCategory : number,
        private description? : string | null,
        private id? : number | null
    ){}

    getId(): number {
    return this.id;
  }

  getEventId(): number {
    return this.eventId;
  }

  getLatitude(): number {
    return this.latitude;
  }

  getLongitude(): number {
    return this.longitude;
  }

  getLabel(): string {
    return this.label;
  }

  getPinnedAt(): Date {
    return this.pinnedAt;
  }

  getPinCategory(): number {
    return this.pinCategory;
  }

  getDescription(): string | null | undefined {
    return this.description;
  }
}